File: ASCMN.m
clear all;
close all;
disp('======================== begin =================================')

all_frame = 0;
all_fixations = 0;

for i = 1:24
    i_video = strcat('raw_data_video', num2str(i));
    i_video = strcat(i_video, '.mat');
    data = load(i_video);
    n_frames = data.nFrames
    all_frame = all_frame + n_frames;
    one_video_data = data.raw_data;
    one_video_subject = length(one_video_data(1, 1, :));
    
    for i_subject = 1:one_video_subject
       one_subject_data = one_video_data(:, :, i_subject);
       
       for i_frame = 1:n_frames 
           one_frame_x = one_subject_data(1, i_frame);
           one_frame_y = one_subject_data(2, i_frame);
           
           if one_frame_x ~= -1 & one_frame_y  ~= -1
               all_fixations = all_fixations + 1;
           end
           
       end
    end
end
all_fixations
all_frame 
ave_fixation = all_fixations / all_frame

File: ASCMN.py
from config import *
from function import *
import scipy.io as sio

def get_all_files_1(path = ASCMN_data_path ):
    f =  get_all_files(path)
    for i in f:
        print("'%s',"%i)

def get_all_frames(path = ASCMN_data_path ):

    all_video_frame = 0
    for i_video in range(1, len(ASCMN_video_list) + 1):
        mat_path = path + 'raw_data_video' + str(i_video) + ".mat"

        data=sio.loadmat(mat_path)
        one_video_data = data["raw_data"]
        one_video_frame = data["nFrames"][0][0]
        all_video_frame += (one_video_frame)
        print(one_video_frame)


    print(">>>> all_video_frame: %d"%all_video_frame)


def get_all_fixations():
    pass


def main():
    get_all_frames()


if __name__ == "__main__":
    main()